### PR TITLE
[WIP] begin with typescript

### LINKED ISSUE #141: Use Flow
use flow. Flow will allow us to continue writing pure JS and add type checking. TypeScript requires us to learn special syntaxes and have it compile back to pure JS.

### PR DESCRIPTION
**Pull request by skv-headless**
closes #141

### CODE REVIEW

File: class/abstract-wallet.js [hunk 1]
```diff
@@ -92,7 +92,7 @@ export class AbstractWallet {
     return true;
   }
 
-  allowSendMax(): boolean {
```
Comment (skv-headless) [inline class/abstract-wallet.js]: What types here for?

Comment (Overtorment) [reply class/abstract-wallet.js]: they help itelij webstorm with autocomplete

File: class/deeplinkSchemaMatch.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import RNFS from 'react-native-fs';
 import url from 'url';
 import { Chain } from '../models/bitcoinUnits';
 const bitcoin = require('bitcoinjs-lib');
-const BlueApp: AppStorage = require('../BlueApp');
```
Comment (Overtorment) [inline class/deeplinkSchemaMatch.js]: whats wrong with typehinting here? its specifically for intelij

Comment (skv-headless) [reply class/deeplinkSchemaMatch.js]: typescript throws an error that types are not allowed in javascript. I can try to shut down this error if needed

Comment (Overtorment) [reply class/deeplinkSchemaMatch.js]: please do, that thing helps with code navigation and autocomplete in IDE

**Comment by Overtorment** [comment]
this PR needs tests to be fixed and conflict resolved

File: screen/settings/about.tsx [hunk 3]
```diff
@@ -135,7 +135,7 @@ const About = () => {
           <BlueTextCentered>
             {getApplicationName()} ver {getVersion()} (build {getBuildNumber()})
           </BlueTextCentered>
-          <BlueTextCentered>{new Date(getBuildNumber() * 1000).toGMTString()}</BlueTextCentered>
+          <BlueTextCentered>{new Date(parseInt(getBuildNumber()) * 1000).toUTCString()}</BlueTextCentered>
```
[the remark this answers is not in the document]
Comment (skv-headless) [reply screen/settings/about.tsx:138]: getBuildNumber returns string. We can't multiple string

[the remark this answers is not in the document]
Comment (Overtorment) [reply screen/settings/about.tsx:138]: image
looks like we can. build number is provided by CI/CD (appcenter.ms) and is always a timestamp

[the remark this answers is not in the document]
Comment (skv-headless) [reply screen/settings/about.tsx:138]: I mean in typescript we can't do it https://www.typescriptlang.org/play/index.html#code/EQCwpgNhD2wAQCo4EYAMqg

here https://github.com/react-native-community/react-native-device-info/blob/master/src/index.ts#L323
return type is string 
`let buildNumber: string;`

File: class/deeplinkSchemaMatch.js (hunk 2 above)
Comment (skv-headless) [reply class/deeplinkSchemaMatch.js:8]: I get this error. Typescript doesn't allow types in js files.
```
screen/wallets/add.js:37:14 - error TS8010: Type annotations can only be used in TypeScript files.
37 let BlueApp: AppStorage = require('../../BlueApp');
```
I didn't find how to suppress specific error in ts and I can't use `//@ts-ignore` in js files.

So the best solution that I found is to rename files with types to ts and ignore all errors there.

**Comment by Overtorment** [comment]
ts support was added in https://github.com/BlueWallet/BlueWallet/pull/3403

**Pull request closed without merging**